kotlin/services/eventbridge/src/main/kotlin/com/kotlin/eventbridge/EventbridgeMVP.kt

```diff
@@ -0,0 +1,729 @@
+        rule = eventRuleName
+    }
+
+    EventBridgeClient { region = "us-east-1" }.use { eventBrClient ->
+        val response = eventBrClient.listTargetsByRule(request)
+        val allTargets = response.targets
+
+        // Get all targets and delete them.
+        if (allTargets != null) {
+            for (myTarget in allTargets) {
+                val removeTargetsRequest = RemoveTargetsRequest {
+                    rule = eventRuleName
+                    ids = listOf(myTarget.id.toString())
+                }
+                eventBrClient.removeTargets(removeTargetsRequest)
+                println("Successfully removed the target")
+            }
+        }
+    }
+}
+// snippet-end:[eventbridge.kotlin.delete.targets.main]
+
+// snippet-start:[eventbridge.kotlin._put_event.main]
+suspend fun triggerCustomRule(email: String) {
+    val json = "{" +
+        "\"UserEmail\": \"" + email + "\"," +
+        "\"Message\": \"This event was generated by example code.\"" +
+        "\"UtcTime\": \"Now.\"" +
+        "}"
+
+    val entry = PutEventsRequestEntry {
+        source = "ExampleSource"
+        detail = json
+        detailType = "ExampleType"
+    }
+
+    val eventsRequest = PutEventsRequest {
+        this.entries = listOf(entry)
+    }
+
+    EventBridgeClient { region = "us-east-1" }.use { eventBrClient ->
+        eventBrClient.putEvents(eventsRequest)
+    }
+}
+// snippet-end:[eventbridge.kotlin._put_event.main]
+
+suspend fun updateCustomRuleTargetWithTransform(topicArn: String?, ruleName: String?) {
+    val targetId = UUID.randomUUID().toString()
+
+    val inputTransformerOb = InputTransformer {
+        inputTemplate = "\"Notification: sample event was received.\""
+    }
+
+    val target = aws.sdk.kotlin.services.eventbridge.model.Target {
+        id = targetId
+        arn = topicArn
+        inputTransformer = inputTransformerOb
+    }
+
+    val targetsRequest = PutTargetsRequest {
+        rule = ruleName
+        targets = listOf(target)
+        eventBusName = null
+    }
+
+    EventBridgeClient { region = "us-east-1" }.use { eventBrClient ->
+        eventBrClient.putTargets(targetsRequest)
+    }
+}
+
+// snippet-start:[eventbridge.kotlin.puttargetstransform.main]
+suspend fun updateToCustomRule(ruleName: String?) {
+    val customEventsPattern = "{" +
+        "\"source\": [\"ExampleSource\"]," +
+        "\"detail-type\": [\"ExampleType\"]" +
+        "}"
+    val request = PutRuleRequest {
+        name = ruleName
+        description = "Custom test rule"
+        eventPattern = customEventsPattern
+    }
+
+    EventBridgeClient { region = "us-east-1" }.use { eventBrClient ->
+        eventBrClient.putRule(request)
+    }
+}
+// snippet-end:[eventbridge.kotlin.puttargetstransform.main]
+
+// Update an Amazon S3 object created rule with a transform on the target.
+suspend fun updateSnsEventRule(topicArn: String?, ruleName: String?) {
+    val targetId = UUID.randomUUID().toString()
```

Comment:
This one looks like a PutTarget example, see above - I think something is mixed up.